src/codeu/chat/database/Database.java

```diff
@@ -0,0 +1,91 @@
+package codeu.chat.database;
+
+import codeu.chat.common.Conversation;
+import codeu.chat.common.Message;
+import codeu.chat.common.User;
+import codeu.chat.util.Uuid;
+import com.mongodb.MongoClientURI;
+import org.bson.Document;
+
+import com.mongodb.MongoClient;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+
+public class Database {
+    public MongoClient mongoClient;
+    public MongoDatabase database;
+    public MongoCollection<Document> users;
+    public MongoCollection<Document> messages;
+    public MongoCollection<Document> conversations;
+
+    private boolean clearDatabase = false;
+    private String username = "User";
+    private String password = "SuperSecretPassword";
+    private String databasePath = "TestDatabase";
+    private String URI = "mongodb://" + username + ":" + password + "@codeu-shard-00-00-kskbs.mongodb.net:27017,codeu-shard-00-01-kskbs.mongodb.net:27017,codeu-shard-00-02-kskbs.mongodb.net:27017/" + databasePath + "?ssl=true&replicaSet=CodeU-shard-0&authSource=admin";
+
+    public Database(){
+        try{
+            MongoClientURI uriConnect = new MongoClientURI(URI);
+            this.mongoClient = new MongoClient(uriConnect);
+            this.database = mongoClient.getDatabase(databasePath);
+            System.out.println("Successful connection to database.");
+
+            if(clearDatabase){
+                removeDatabaseCollections();
+            }
+
+            createDatabaseCollections();
+
+        }catch(Exception e){
+            System.err.println( e.getClass().getName() + ": " + e.getMessage() );
```

Comment:
We should try to eliminate the pattern of dealing with exceptions this way as much as we can. The biggest problem is that after a problem we just continue executing normally as if nothing happened; yet we have a Database instance that won't work.

Actually, to pull back a bit, we should think about whether each exception is *recoverable* or not. If it's an unexpected situation and it means everything is broken and bad, we should rethrow as an unchecked exception. (I'd define a very simple class `UnrecoverableException extends RuntimeException` for this purpose and reuse it often. Always chain the original exception in.)

If there's some action you can take to "make things right" that's when to catch. If you think the *caller* of the method may have a useful action to "make things right", that's when to let a checked exception pass out. You may still want to change the *type* of exception thrown to one that makes sense for this particular class and its caller. (Construct a new exception, passing/"chaining" the existing one in.)